Intentional or bug? : src/tests/stdio/printf_test.c:122:23: error: ‘snprintf’ output truncated before the last format character [-Werror=format-truncation=]

```
gcc  -pipe --debug-prefix-map=`pwd`= --debug-prefix-map=./= -fno-strict-aliasing -fno-common -Wall -Werror -Wundef -Wno-trigraphs -Wno-char-subscripts -Wformat -std=gnu99 -O0 -gdwarf-2 -nostdinc -m32 -march=i386 -fno-stack-protector -Wno-array-bounds -Wno-misleading-indentation -Wno-nonnull-compare -U__linux__ -Ulinux -U__linux -D__EMBOX__ -D__unix -imacros build/base/gen/config.lds.h -Ibuild/base/include -I./src/include -I./src/arch/x86/include -Ibuild/base/gen/include -Ibuild/base/gen/src/include -Ibuild/base/gen/src/include -Ibuild/base/gen/src/arch/x86/include  -I./src/compat/linux/include -I./src/compat/posix/include -I./src/compat/libc/include -I/home/agu/_local/p_c/github_embox_embox/src/compat/cxx/include -nostdinc -MMD -MP  -include build/base/gen/include/config/embox/test/stdio/printf_test.h -D__EMBUILD_MOD__=embox__test__stdio__printf_test -c -o build/base/obj/src/tests/stdio/printf_test.o src/tests/stdio/printf_test.c
In file included from ./src/include/embox/test.h:32,
                 from src/tests/stdio/printf_test.c:10:
src/tests/stdio/printf_test.c: In function ‘__test_case_at_line_113’:
src/tests/stdio/printf_test.c:122:23: error: ‘snprintf’ output truncated before the last format character [-Werror=format-truncation=]
  test_assert_equal(4, snprintf(dest, 4, "1234"));
./src/include/framework/test/assert.h:129:39: note: in definition of macro ‘test_assert_equal’
  __test_assertion_handle((actual) == (expected), \
                                       ^~~~~~~~
src/tests/stdio/printf_test.c:122:23: note: ‘snprintf’ output 5 bytes into a destination of size 4
  test_assert_equal(4, snprintf(dest, 4, "1234"));
./src/include/framework/test/assert.h:129:39: note: in definition of macro ‘test_assert_equal’
  __test_assertion_handle((actual) == (expected), \
                                       ^~~~~~~~
src/tests/stdio/printf_test.c:125:23: error: ‘12345678’ directive output truncated writing 8 bytes into a region of size 2 [-Werror=format-truncation=]
  test_assert_equal(8, snprintf(dest, 2, "12345678"));
./src/include/framework/test/assert.h:129:39: note: in definition of macro ‘test_assert_equal’
  __test_assertion_handle((actual) == (expected), \
                                       ^~~~~~~~
src/tests/stdio/printf_test.c:125:23: note: ‘snprintf’ output 9 bytes into a destination of size 2
  test_assert_equal(8, snprintf(dest, 2, "12345678"));
./src/include/framework/test/assert.h:129:39: note: in definition of macro ‘test_assert_equal’
  __test_assertion_handle((actual) == (expected), \
                                       ^~~~~~~~
cc1: all warnings being treated as errors
make[4]: *** [mk/image_lib.mk:33: build/base/obj/src/tests/stdio/printf_test.o] Error 1
make[3]: *** [mk/build.mk:23: build] Error 2
make[2]: *** [mk/load.mk:41: build] Error 2
make[1]: *** [mk/main.mk:30: build] Error 2
make: *** [Makefile:37: all] Error 2
```

```
src/tests/stdio/printf_test.c:122:23: error: ‘snprintf’ output truncated before the last format character [-Werror=format-truncation=]
  test_assert_equal(4, snprintf(dest, 4, "1234"));
```
gcc issues warning as the result (in `dest`) do not hold all required bytes, deduced in static.
But the test is actually checks if snprintf works correctly in such situation. 
I think disabling the warning for this file with `#pragma`s would be good enough solution

--- a/src/tests/mem/heap.c
+++ b/src/tests/mem/heap.c
@@ -142,5 +142,16 @@ TEST_CASE("Allocates many objects with different size") {
 
 TEST_CASE("malloc fails when trying to allocate a very large"
 		" chunk of memory") {
+#ifdef __GNUC__
+#if __GNUC__ > 6
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Walloc-size-larger-than="
+#endif
+#endif
 	test_assert_null(malloc(4294966160));
+#ifdef __GNUC__
+#if __GNUC__ > 6
+#pragma GCC diagnostic pop
+#endif
+#endif
 }

--- a/src/tests/stdio/printf_test.c
+++ b/src/tests/stdio/printf_test.c
@@ -118,11 +118,20 @@ TEST_CASE("Test of snprintf with truncated output") {
 
 	test_assert_equal(3, snprintf(&random_char, 0, "012"));
 	test_assert_equal(random_char, backup_char);
-
+#ifdef __GNUC__
+#if __GNUC__ > 6
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
+#endif
+#endif
 	test_assert_equal(4, snprintf(dest, 4, "1234"));
 	test_assert_str_equal(dest, "123");
-
 	test_assert_equal(8, snprintf(dest, 2, "12345678"));
+#ifdef __GNUC__
+#if __GNUC__ > 6
+#pragma GCC diagnostic pop
+#endif
+#endif
 	test_assert_str_equal(dest, "1");
 }
 
@@ -134,7 +143,7 @@ TEST_CASE("Test of printing with mistake in format") {
 }
 
 #if 0
-/* FIXME this test should respect support_floating option of 
+/* FIXME this test should respect support_floating option of
  * embox.compat.libc.stdio.print. Until then, disabled
  */
 TEST_CASE("Test of specifier with type float") {

